core_impl/src/test/java/com/google/instrumentation/stats/StatsManagerImplTest.java

```diff
@@ -192,16 +199,26 @@ public void testRecordNonExistentMeasurementDescriptor() {
     assertThat(view.getDistributionAggregations()).hasSize(0);
   }
 
-  // TODO(songya): update this test once we determine how to handle tags that aren't an exact match
-  // for the view.
+
   @Test
   public void testRecordNonExistentTag() {
     statsManager.registerView(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW);
-    statsManager.record(
-        wrongTag, MeasurementMap.of(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 10.0));
+    statsManager.record(wrongTag, MeasurementMap.of(
+        RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 10.0));
+    statsManager.record(wrongTag, MeasurementMap.of(
+        RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 50.0));
     DistributionView view =
         (DistributionView) statsManager.getView(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW);
-    // Won't record stats if there are non existent tags.
-    assertThat(view.getDistributionAggregations()).hasSize(0);
+
+    DistributionAggregation distributionAggregation = view.getDistributionAggregations().get(0);
+    assertThat(view.getDistributionAggregations()).hasSize(1);
+    List<Tag> tags = distributionAggregation.getTags();
+    // Won't record the unregistered tag key, will use default tag instead:
+    // "method" : "unknown/not set".
+    assertThat(tags.get(0).getKey()).isNotEqualTo(wrongTagKey);
```

Comment:
> assertThat(tags.get(0).getKey()).isNotEqualTo(wrongTagKey);

I think this assertion is redundant when we also have `assertThat(tags.get(0).getKey()).isEqualTo(tagKey);`.